From 7542913698830f465c83be88652c45b3987ad5fa Mon Sep 17 00:00:00 2001
From: Yaroslav Zviezda <acroreiser@gmail.com>
Date: Sun, 16 Jul 2023 13:38:58 +0300
Subject: [PATCH] batterysaver: add property to disable night mode on
 battery saver

Night theme on battery saver mode is useful on AMOLED panels
but does not provide benefits for IPS. To avoid runtime overhead of theme switch
we can just keep current theme by setting "ro.config.night_mode_on_battery_saver" to "false".

Change-Id: I4800d0d3aa3965bb8fbd5301482061dc8153cda6
---
 .../android/server/power/batterysaver/BatterySaverPolicy.java  | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/services/core/java/com/android/server/power/batterysaver/BatterySaverPolicy.java b/services/core/java/com/android/server/power/batterysaver/BatterySaverPolicy.java
index b4613a76..715a77bc 100644
--- a/services/core/java/com/android/server/power/batterysaver/BatterySaverPolicy.java
+++ b/services/core/java/com/android/server/power/batterysaver/BatterySaverPolicy.java
@@ -26,6 +26,7 @@ import android.os.Handler;
 import android.os.PowerManager;
 import android.os.PowerManager.ServiceType;
 import android.os.PowerSaveState;
+import android.os.SystemProperties;
 import android.provider.DeviceConfig;
 import android.provider.Settings;
 import android.text.TextUtils;
@@ -499,7 +500,7 @@ public class BatterySaverPolicy extends ContentObserver implements
                 rawPolicy.enableDataSaver,
                 rawPolicy.enableFirewall,
                 // Don't force night mode when car projection is enabled.
-                rawPolicy.enableNightMode && !mAutomotiveProjectionActive.get(),
+                rawPolicy.enableNightMode && !mAutomotiveProjectionActive.get() && SystemProperties.getBoolean("ro.config.night_mode_on_battery_saver", true),
                 rawPolicy.enableQuickDoze,
                 rawPolicy.forceAllAppsStandby,
                 rawPolicy.forceBackgroundCheck,
-- 
2.43.0

